sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -393,6 +393,42 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * SPARK-28443: Spark sql add exception when create field type NullType
+ */
+object DDLCheck extends (LogicalPlan => Unit) {
+
+  def failAnalysis(msg: String): Unit = { throw new AnalysisException(msg) }
+
+  override def apply(plan: LogicalPlan): Unit = {
+    plan.foreach {
+      case ct: CreateTable if ct.tableDesc.schema.exists { f =>
+        DataTypes.NullType.sameType(f.dataType)
+      } =>
+        failAnalysis("DataType NullType is not supported for create table ")
+
+      case ct2: CreateV2Table if ct2.tableSchema.exists { f =>
+        DataTypes.NullType.sameType(f.dataType)
+      } =>
+        failAnalysis("DataType NullType is not supported for create table")
+
+      // DataSourceStrategy will convert CreateTable to CreateDataSourceTableCommand before check
+      case cdstc: CreateDataSourceTableCommand if cdstc.table.schema.exists { f =>
```

Comment:
I know this isn't much code, but I think it would be better to refactor this `exists` and the `failAnalysis` call into a method so it isn't repeated several times.

Reply:
Thanks for your advise.